pkgs/development/python-modules/tensorflow/default.nix

```diff
@@ -299,9 +302,21 @@ let
 
     TF_NEED_CUDA = tfFeature cudaSupport;
     TF_CUDA_PATHS = lib.optionalString cudaSupport "${cudatoolkit_joined},${cudnn},${nccl}";
+    TF_CUDA_COMPUTE_CAPABILITIES = builtins.concatStringsSep "," cudaCapabilities;
+
+    # Toolchain for cuda-enabled builds.
+    # Tensorflow looks at many related variables which may diverge.
+    # We want to achieve two things:
+    # 1. NVCC should use a compatible back-end (e.g. gcc11 for cuda11)
+    # 2. Normal C++ files should be compiled with the same toolchain,
+    #    since otherwise we sometimes get "`GLIBCXX_3.4.30' not found"
+    #    at runtime.
     GCC_HOST_COMPILER_PREFIX = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin";
-    GCC_HOST_COMPILER_PATH = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin/gcc";
-    TF_CUDA_COMPUTE_CAPABILITIES = builtins.concatStringsSep "," cudaFlags.cudaRealArches;
+    GCC_HOST_COMPILER_PATH = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin/cc";
+    HOST_C_COMPILER = lib.optionalString cudaSupport "${cudatoolkit_cc_joined}/bin/cc";
```

Comment:
Alternative: override `buildPythonPackage`'s `stdenv`, as in https://github.com/SomeoneSerge/nixpkgs/commit/cc4f01552c2dca50b452170df2770edb71148555

I'd rather keep this for later because it seems like a potentially big change